qcodes_contrib_drivers/drivers/QDevil/QDAC2.py

```diff
@@ -1815,6 +1815,7 @@ def __init__(self, name: str, address: str, **kwargs) -> None:
             **kwargs: additional argument to the Visa driver
         """
         super().__init__(name, address, terminator='\n', **kwargs)
+        self._check_instrument_name(name)
```

Comment:
It would probably be better to do this before the call to `__init__` otherwise you will leave around a partially setup instrument.